starlette_context/middleware/context_middleware.py

```diff
@@ -43,15 +49,22 @@ async def set_context(self, request: Request) -> dict:
     async def dispatch(
         self, request: Request, call_next: RequestResponseEndpoint
     ) -> Response:
-        _starlette_context_token: Token = _request_scope_context_storage.set(
-            await self.set_context(request)
-        )
+        try:
+            context = await self.set_context(request)
+            token: Token = _request_scope_context_storage.set(context)
+        except ValueError as e:
+            if isinstance(e.args[0], Response):
+                error_response = e.args[0]
```

Comment:
That's hacky. As I said, I'd prefer if the error message was created in a single place, in the class, something like `handle_plugin_error(self, exc, request)`. wdyt?

Reply:
yes, this part is quite hacky indeed. My original attempt was to define the error response on the plugin so the error message could be fine-grained, before moving back up to have the default_error_response at the middleware level.
Fundamentally, the options for the plugin to communicate back to the middleware are quite limited, so I opted to put the response itself as Exception argument.
It would make sense to define it from the middleware exclusively, but that would tend to enforce a generic catch-all error response. This allows both fine-grained and generic approaches.
Passing a callback function might also be a possibility, but it makes it slightly more annoying to implement on user side, and might feel like a small re-implementation of the error handling system.